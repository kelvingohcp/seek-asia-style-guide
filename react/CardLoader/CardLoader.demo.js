import React from 'react';
import PropTypes from 'prop-types';
import CardLoader from './CardLoader';
import CardLoadingBar from '../CardLoadingBar/CardLoadingBar';
import PageBlock from '../PageBlock/PageBlock';

const cardLoadingBars = [
  (<div style={{ display: 'flex' }} key={1}>
    <div style={{ width: '30%' }}>
      <CardLoadingBar width={'80%'} height={5} />
    </div>
    <div style={{ width: '35%' }}>
      <CardLoadingBar width={40} />
      <CardLoadingBar />
      <CardLoadingBar width={'30%'} />
      <CardLoadingBar />
      <CardLoadingBar width={'30%'} />
    </div>
    <div style={{ width: '35%' }}>
      <CardLoadingBar width={40} inverse />
      <CardLoadingBar />
      <CardLoadingBar width={'30%'} inverse />
      <CardLoadingBar />
      <CardLoadingBar width={20} inverse />
    </div>
  </div>),
  (<CardLoadingBar key={2} />),
  (<CardLoadingBar width={'100%'} key={3} />)
];

const CardLoaderContainer = ({ component: DemoComponent, componentProps }) => {
  return (
    <PageBlock style={{ width: '100%' }}>
      <DemoComponent {...componentProps} >
        {cardLoadingBars}
      </DemoComponent>
    </PageBlock>
  );
};

CardLoaderContainer.propTypes = {
  component: PropTypes.element,
  componentProps: PropTypes.object
};

export default {
  route: '/CardLoader',
  title: 'Card Loader',
  component: CardLoader,
  container: CardLoaderContainer,
  initialProps: {
    children: cardLoadingBars
  },
  options: []
};
