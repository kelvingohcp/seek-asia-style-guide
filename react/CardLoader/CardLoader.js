import React from 'react';
import Card from '../Card/Card';
import styles from './CardLoader.less';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const CardLoader = ({ children, className }) => {
  return (
    <Card className={classnames(styles.root, className)}>
      <div className={styles.animatedBackground}>
        {children}
      </div>
    </Card>
  );
};

CardLoader.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};

export default CardLoader;
