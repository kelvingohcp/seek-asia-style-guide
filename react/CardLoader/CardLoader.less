@import (reference) "~seek-asia-style-guide/theme";

.root {
  background: @saWhite;
  width: 100%;
  padding: @grid-gutter-width;
}

.animatedBackground {
  animation-duration: 1s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: placeHolderShimmer;
  animation-timing-function: linear;
  background: @sk-gray-lightest;
  background: linear-gradient(75deg, @sk-gray-light 8%, @sk-mid-gray-light 18%, @sk-gray-light 33%);
  background-size: 200% 100%;
  position: relative;
}

@keyframes placeHolderShimmer {
  0% {
    background-position: 95% 0%;
  }
  100% {
    background-position: -95% 0%;
  }
}
