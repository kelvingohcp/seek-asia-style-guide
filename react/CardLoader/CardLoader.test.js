import React from 'react';
import { shallow } from 'enzyme';

import CardLoader from './CardLoader';

describe('CardLoader', () => {
  it('should render with defaults', () => {
    expect(shallow(<CardLoader />)).toMatchSnapshot();
  });
});
