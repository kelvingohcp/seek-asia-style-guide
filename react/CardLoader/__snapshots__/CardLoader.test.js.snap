// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardLoader should render with defaults 1`] = `
<Card
  className="root"
>
  <div
    className="animatedBackground"
  />
</Card>
`;
